packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -1432,11 +1442,14 @@ export class UserPool extends UserPoolBase {
     if (props.allowedFirstAuthFactors.passkey) {
       allowedFirstAuthFactors.push('WEB_AUTHN');
     }
-    if (allowedFirstAuthFactors.length === 1) {
-      return undefined;
-    }
 
-    if (props.featurePlan === FeaturePlan.LITE) {
+    /*
+     * Choice-based authentication is enabled when built allowedFirstAuthFactors contains any factor but PASSWORD.
+     * This check should be placed here to supply the way to disable choice-based authentication explicitly
+     * by specifying `allowedFirstAuthFactors: { password: true }`.
+     */
+    const isChouseBasedAuthenticationEnabled = allowedFirstAuthFactors.some((auth) => auth !== 'PASSWORD');
+    if (isChouseBasedAuthenticationEnabled && props.featurePlan === FeaturePlan.LITE) {
       throw new ValidationError('To enable choice-based authentication, set `featurePlan` to `FeaturePlan.ESSENTIALS` or `FeaturePlan.PLUS`.', this);
```

Comment:
```suggestion
    if (props.featurePlan === FeaturePlan.LITE) {
      throw new ValidationError('To enable choice-based authentication, set `featurePlan` to `FeaturePlan.ESSENTIALS` or `FeaturePlan.PLUS`.', this);
```

Reply:
`allowedFirstAuthFactors: { password: true }` must allow Lite feature plan.

When the user have a user pool with choice-based authentication enabled and want to downgrade to Lite feature plan, the user must specify `allowedFirstAuthFactors: { password: true }` to disable choice-based authentication explicitly.
see https://github.com/aws/aws-cdk/pull/32369#discussion_r1938673153